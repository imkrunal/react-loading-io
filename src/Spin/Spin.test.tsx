import React from 'react'
import renderer from 'react-test-renderer'
import { Spin } from './'

it('renders correctly', (): void => {
  const tree = renderer.create(<Spin />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('joins default class with className prop', (): void => {
  const { root } = renderer.create(<Spin className='spinner' />)

  expect(root.findByType('div').props.className).toEqual('rl-spin spinner')
})

it('sets CSS vars with props', (): void => {
  const size = 100

  const { root } = renderer.create(
    <Spin size={size} color='#d36ac2' speed={2} style={{ margin: 20 }} />
  )
  const { style } = root.findByType('div').props

  const scale = size / 200
  const translate = size * scale

  expect(style).toEqual({
    '--rl-spin-size': '100px',
    '--rl-spin-color': '#d36ac2',
    '--rl-spin-scale': scale,
    '--rl-spin-translate': `${translate}px`,
    '--rl-spin-speed': `2s`,
    '--rl-spin-border': '40px',
    margin: 20
  })
})
