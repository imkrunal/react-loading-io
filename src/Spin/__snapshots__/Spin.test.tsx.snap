// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rl-spin"
  style={
    Object {
      "--rl-spin-border": "40px",
      "--rl-spin-color": "#ED4585",
      "--rl-spin-scale": 1,
      "--rl-spin-size": "200px",
      "--rl-spin-speed": "1s",
      "--rl-spin-translate": "200px",
    }
  }
>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
</div>
`;
